content/blog/posts/vectors-matrices-matrix-order.md

```diff
@@ -0,0 +1,733 @@
+    {3.f, 0.f, 3.f, 4.f}  // column 2
+};
+
+// 4x1 column vector to hold the result
+float result[rows] = {0.f, 0.f, 0.f, 0.f};
+
+for(std::size_t r = 0; r < rows; r++)
+{
+   for(std::size_t c = 0; c < cols; c++)
+   {
+      result[r] += columnMajor[c][r] * columnVector[c];
+   }
+}
+
+// prints 4 0 4 8
+for(std::size_t i = 0; i < rows; i++)
+{
+   std::cout << result[i] << " ";
+}
+```
+
+You may have noticed that the code for both snippets is identical, bar some flipping of variable names for clarity. Remember that the product of a row vector with a given matrix is the transpose of the product between a column vector and the transpose of that same matrix:
+
+$$
+\vec{r} \cdot \mathbf{M} = (\mathbf{M^T} \cdot \vec{c})^\mathbf{T}}
+$$
+
+An interesting property of storing our matrices in the array matrix order convention of our vector convention (i.e. row vectors with matrices stored in row-major order and column vectors with matrices stored in column-major order) is that the actual array data for the matrices is identical regardless of which of the two convention combinations you choose. Consider the following 2d transform matrix for scaling, anti-clockwise rotating and translating a given row vector:
+
+$$
+\begin{bmatrix}      S_{x} \cdot \cos(\theta) & \sin(\theta) & 0  \\\      -\sin(\theta) & S_{y} \cdot \cos(\theta) & 0  \\\  T_{x} &  T_{y}  & 1  \end{bmatrix}
+$$
+
+When laid out in row-major in a two dimensional array, the array is initialized as follows:
+
+```c++
+// Scale to double and quadruple the x and y axis accordingly
+const float Sx = 2.f;
+const float Sy = 4.f
+
+// Rotation of approx. 45 degrees (in radians)
+const float theta = 0.707;
+
+// Translation of -5 along x axis and 10 along y axis
+const float Tx = -5.f;
+const float Ty = 10.f;
+
+// 3x3 matrix in row-major order
+const std::size_t rows = 3;
+const std::size_t cols = 3;
+
+const float rowMajor[rows][cols] = 
+{ 
+    {Sx * cos(theta), sin(theta),      0.f}, // row 0 
+    {-sin(theta),     Sy * cos(theta), 0.f}, // row 1
+    {Tx,              Ty,              1.f}  // row 2
+};
+```
+
+Now consider the same transformation matrix transposed for translating a given column vector:
+
+$$
+\begin{bmatrix}      S_{x} \cdot \cos(\theta) & -\sin(\theta) & T_{x} \\\      \sin(\theta) & S_{y} \cdot \cos(\theta) & T_{y} \\\      0 & 0 & 1  \end{bmatrix}
+$$
+
+When laid out in column-major in a two dimensional array, the array is initialized as follows:
+
+```c++
+// Scale to double and quadruple the x and y axis accordingly
+const float Sx = 2.f;
+const float Sy = 4.f
+
+// Rotation of approx. 45 degrees (in radians)
+const float theta = 0.785;
+
+// Translation of -5 along x axis and 10 along y axis
+const float Tx = -5.f;
+const float Ty = 10.f;
+
+// 3x3 matrix in column-major order
+const std::size_t rows = 3;
+const std::size_t cols = 3;
+
+const float columnMajor[cols][rows] = 
+{ 
+    {Sx * cos(theta), sin(theta),      0.f}, // column 0 
```

Comment:
Same comment about the scale in 1,2 and 2,1 elements.

Reply:
Fixed 👍